modules/indexer/issues/meilisearch/meilisearch.go

```diff
@@ -241,3 +246,60 @@ func parseSortBy(sortBy internal.SortBy) string {
 	}
 	return field + ":asc"
 }
+
+// nonFuzzyWorkaround is needed as meilisearch does not have an exact search
+// and you can only change "typo tolerance" per index. So we have to post-filter the results
+// https://www.meilisearch.com/docs/learn/configuration/typo_tolerance#configuring-typo-tolerance
+// TODO: remove once https://github.com/orgs/meilisearch/discussions/377 is addressed
+func nonFuzzyWorkaround(searchRes *meilisearch.SearchResponse, keyword string, isFuzzy bool) ([]internal.Match, error) {
+	hits := make([]internal.Match, 0, len(searchRes.Hits))
+	for _, hit := range searchRes.Hits {
+		hit, ok := hit.(map[string]any)
+		if !ok {
+			return nil, ErrMalformedResponse
+		}
+
+		if !isFuzzy {
+			keyword = strings.ToLower(keyword)
+			title, ok := hit["title"].(string)
+			if !ok {
+				return nil, ErrMalformedResponse
+			}
+			if !strings.Contains(strings.ToLower(title), keyword) {
+				content, ok := hit["content"].(string)
+				if !ok {
+					return nil, ErrMalformedResponse
+				}
+				if !strings.Contains(strings.ToLower(content), keyword) {
+					comments, ok := hit["comments"].([]any)
+					if !ok {
+						return nil, ErrMalformedResponse
+					}
+					found := false
+					for i := range comments {
+						comment, ok := comments[i].(string)
+						if !ok {
+							return nil, ErrMalformedResponse
+						}
+						if strings.Contains(strings.ToLower(comment), keyword) {
+							found = true
+							break
+						}
+					}
+					if !found {
+						// we could not have a direct match, so ignore that hit and move on ...
+						continue
+					}
+				}
+			}
+		}
```

Comment:
Can something be done to make this more "flat"?

Reply:
not without loosing performance or over-engineering it

Reply:
PS: If you have a good idea, I'll apply it :)

Reply:
@silverwind do you like this aproach more:

```go
		if !isFuzzy {
			keyword = strings.ToLower(keyword)
			match := false

			for _, fn := range []func() (bool, error){
				func() (bool, error) {
					title, ok := hit["title"].(string)
					if !ok {
						return false, ErrMalformedResponse
					}
					return strings.Contains(strings.ToLower(title), keyword), nil
				},
				func() (bool, error) {
					content, ok := hit["content"].(string)
					if !ok {
						return false, ErrMalformedResponse
					}
					return strings.Contains(strings.ToLower(content), keyword), nil
				},
				func() (bool, error) {
					comments, ok := hit["comments"].([]any)
					if !ok {
						return false, ErrMalformedResponse
					}
					for i := range comments {
						comment, ok := comments[i].(string)
						if !ok {
							return false, ErrMalformedResponse
						}
						if strings.Contains(strings.ToLower(comment), keyword) {
							return true, nil
						}
					}
					return false, nil
				},
			} {
				if contain, err := fn(); err != nil {
					return nil, err
				} else if contain {
					match = true
					break
				}
			}

			if !match {
				continue
			}
		}
```

it is **flat** but I think not everybody will understand

Reply:
I can see how it works, but this nonlinear flow is arguable worse than before I think. I guess go just isn't made to work with deeply nested maps.

Reply:
@silverwind I optimized and simplified the idea above ... now it should be flat and normal maintainers should understand